File: app/src/main/java/com/example/tvprogramsscanner/other/MyRunnable.java
package com.example.tvprogramsscanner.other;

import com.example.tvprogramsscanner.service.RequestService;

import java.io.IOException;

public class MyRunnable implements Runnable {

    RequestService requestService;
    String serialNumber;
    int borderId, direction;

    String jsonResponseString = null;

    public MyRunnable(RequestService service, String serialNumber, int borderId, int direction){
        this.requestService = service;
        this.serialNumber = serialNumber;
        this.borderId = borderId;
        this.direction = direction;
    }

    @Override
    public void run() {
        try {
            requestService.makeRequest(serialNumber, borderId, direction);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public String getJsonResponseString() {
        return jsonResponseString;
    }
}

File: app/src/main/java/com/example/tvprogramsscanner/other/MyRunnable.kt
package com.example.tvprogramsscanner.other

import com.example.tvprogramsscanner.service.RequestService
import java.io.IOException

class MyRunnable(var requestService: RequestService, var serialNumber: String,
    var borderId: Int, var direction: Int) : Runnable {

    var jsonResponseString: String? = null

    override fun run() {
        try {
            requestService.makeRequest(serialNumber, borderId, direction)
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }
}